kvbc/test/sparse_merkle_storage/db_adapter_unit_test.cpp

```diff
@@ -212,34 +223,132 @@ TEST_P(db_adapter_custom_blockchain, add_block_with_non_provable_keys) {
+  for (size_t i = 0u; i < updates.size(); ++i) {
+    const auto &[value, blockId] = adapter.getValue(updates[i].begin()->first, adapter.getLastReachableBlockId());
+    ASSERT_EQ(blockId, i + 1);
+    ASSERT_EQ(value, updates[i].begin()->second);
+  }
+
+  ASSERT_THROW(adapter.getValue(Sliver{"non-exisiting-key"}, adapter.getLastReachableBlockId()), NotFoundException);
+
+  // Make sure that blocks with non-provable keys does not affect the Merkle Tree state
+  ASSERT_EQ(initialTreeVersion, adapter.getStateVersion());
+  ASSERT_EQ(initialHash, adapter.getStateHash());
+
+  // Make sure that adding blocks with provable keys affects the Merkle Tree state
+  adapter.addBlock(SetOfKeyValuePairs{{Sliver{"provable-key"}, Sliver{"provable-value"}}});
+  ASSERT_EQ(initialTreeVersion + 1, adapter.getStateVersion());
+  ASSERT_NE(initialHash, adapter.getStateHash());
+}
+
+TEST_P(db_adapter_custom_blockchain, raw_block_with_non_provable_and_provable_keys) {
+  const Sliver skey1{"skey1"};
+  const Sliver skey2{"skey2"};
+  const Sliver skey3{"skey3"};
+
+  const Sliver svalue1{"svalue1"};
+  const Sliver svalue2{"svalue2"};
+  const Sliver svalue3{"svalue3"};
+
+  const Sliver key1{"key1"};
+  const Sliver key2{"key2"};
+  const Sliver key3{"key3"};
+
+  const Sliver value1{"value1"};
+  const Sliver value2{"value2"};
+  const Sliver value3{"value3"};
+
+  auto adapter = DBAdapter{GetParam()->newEmptyDb(), true, {skey1, skey2, skey3}};
+
+  std::vector<SetOfKeyValuePairs> updates = {
+      {{skey1, svalue1}}, {{skey2, svalue2}}, {{skey3, svalue3}}, {{key1, value1}}, {{key2, value2}}, {{key3, value3}}};
+  for (const auto &u : updates) {
+    adapter.addBlock(u);
+  }
+
+  for (size_t i = 0u; i < updates.size(); ++i) {
+    ASSERT_EQ(block::detail::getData(adapter.getRawBlock(i + 1)), updates[i]);
+  }
+}
+
+TEST_P(db_adapter_custom_blockchain, raw_block_only_with_non_provable_keys) {
+  const Sliver skey1{"key1"};
+  const Sliver skey2{"key2"};
+  const Sliver skey3{"key3"};
+
+  const Sliver svalue1{"value1"};
+  const Sliver svalue2{"value2"};
+  const Sliver svalue3{"value3"};
+
+  auto adapter = DBAdapter{GetParam()->newEmptyDb(), true, {skey1, skey2, skey3}};
+
+  std::vector<SetOfKeyValuePairs> updates = {{{skey1, svalue1}}, {{skey2, svalue2}}, {{skey3, svalue3}}};
+  for (const auto &u : updates) {
+    adapter.addBlock(u);
   }
 
-  ASSERT_THROW(adapter.getValue(defaultSliver, numBlocks), NotFoundException);
+  for (size_t i = 0u; i < updates.size(); ++i) {
+    ASSERT_EQ(block::detail::getData(adapter.getRawBlock(i + 1)), updates[i]);
+  }
+}
+
+TEST_P(db_adapter_custom_blockchain, raw_block_only_with_provable_keys) {
+  const Sliver key1{"key1"};
+  const Sliver key2{"key2"};
+  const Sliver key3{"key3"};
+
+  const Sliver value1{"value1"};
+  const Sliver value2{"value2"};
+  const Sliver value3{"value3"};
+
+  auto adapter = DBAdapter{GetParam()->newEmptyDb(), true};
```

Comment:
Do we need to configure a set of non-provable keys in this test and then verify that provable ones work? What I am getting at is that the case without non-provable keys (default empty param) is covered in other tests.

Reply:
Added the set of non-provable keys.